### PR TITLE
cmake: set big-endian flags to TOOLCHAIN_*_FLAGS

### PR DESCRIPTION
**Pull request by ntavish**
This PR is related to https://github.com/zephyrproject-rtos/sdk-ng/pull/765

### CODE REVIEW

File: cmake/compiler/gcc/target_arm.cmake [hunk 1]
```diff
@@ -10,6 +10,11 @@ endif()
 
 list(APPEND ARM_C_FLAGS -mabi=aapcs)
 
+if(CONFIG_BIG_ENDIAN)
```
Comment (tejlmand) [inline cmake/compiler/gcc/target_arm.cmake:13]: currently `BIG_ENDIAN` is a promptless symbol, meaning it depends on something having a `select BIG_ENDIAN` (or a change of default).

Currently the arch for which that is done is `SPARC`:
https://github.com/zephyrproject-rtos/zephyr/blob/2457f730f43079d8d0626f6dee6f51765a29605d/arch/Kconfig#L65
https://github.com/zephyrproject-rtos/zephyr/blob/2457f730f43079d8d0626f6dee6f51765a29605d/arch/Kconfig#L70

so how is this expected to be selected for `arm v7-r` and thus selecting this code path ?

**Review by tejlmand — CHANGES_REQUESTED** [review]
Seems ok, but some small clarification is needed as this doesn't seem to really add support for big endian on v7-r.

Also this help text needs an update, as it doesn't include the flags on the compiler.
https://github.com/zephyrproject-rtos/zephyr/blob/2457f730f43079d8d0626f6dee6f51765a29605d/arch/Kconfig#L183-L184

**Comment by ntavish** [comment]
@tejlmand sorry for the delay, made the changes you have requested for `zephyr/arch/Kconfig`.

**Comment by tejlmand** [comment]
> @tejlmand sorry for the delay, made the changes you have requested for `zephyr/arch/Kconfig`.

Sorry, but I don't see any reply to this comment: https://github.com/zephyrproject-rtos/zephyr/pull/73563#discussion_r1665666538

**Comment by ntavish** [comment]
> Sorry, but I don't see any reply to this comment: [#73563 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/73563#discussion_r1665666538)

Oops, it was still 'pending', should be visible now.

**Review by tejlmand — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Pull request merged**